src/App.tsx

```diff
@@ -17,18 +58,40 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={searchQuery}
+                searchQuery={setSearchQuery}
+                filter={filter}
+                filterTodo={setFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0
+                ? (<Loader />)
+                : (
+                  <TodoList
+                    todos={visibleTodo}
+                    selectedTodoId={todoId}
+                    selectTodo={(value: number) => {
+                      setTodoId(value);
+                    }}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {todoId !== 0 && (
+        <TodoModal
+          todoId={todoId}
+          todos={todos}
+          selectTodo={(value: number) => {
+            setTodoId(value);
+          }}
```

Comment:
Todo modal no need all todos and selected todo id.

just find selected Todo and pass it to modal
```suggestion
        <TodoModal
          selectedTodo={selectedTodo}
          onClose={(value: number) => setTodoId(0)}
```